evalml/objectives/binary_classification_objective.py

```diff
@@ -43,7 +43,7 @@ def cost(threshold):
             cost = self.objective_function(y_true, y_predicted, X=X)
             return -cost if self.greater_is_better else cost
 
-        optimal = minimize_scalar(cost, method="Golden", options={"maxiter": 100})
+        optimal = minimize_scalar(cost, bounds=(0, 1), method="Bounded")
```

Comment:
Using `bounded` rather than `Golden` in order to ensure that our threshold is between 0 and 1. `Bounded` uses a similar technique to `Brent`, which is preferable to `Golden`, according to [their docs](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.minimize_scalar.html#scipy.optimize.minimize_scalar)

Reply:
@bchen1116 please add a second entry to the release notes for this, under "Enhancements": "Updated threshold optimization method for binary classification"